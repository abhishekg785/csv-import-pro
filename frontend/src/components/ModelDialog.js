import React from 'react'

import { Modal as MuiModel, Typography, withStyles } from '@material-ui/core'

type Props = {
  title?: string,
  isModalOpen?: boolean,
  children: any,
  closeModal?: Function,
  classes: any,
}

const styles = theme => ({
  paper: {
    position: 'absolute',
    left: '25%',
    top: '15%',
    width: theme.spacing.unit * 50,
    backgroundColor: theme.palette.background.paper,
    boxShadow: theme.shadows[5],
    padding: theme.spacing.unit * 4,
  },
});

function ModelDialog({
  title,
  isModalOpen,
  children,
  closeModal,
  classes,
}: Props) {
  return (
    <MuiModel
      open={isModalOpen}
      onClose={closeModal}
    >
      <div className={classes.paper}>
        <Typography>{title}</Typography>
        {children}
      </div>
    </MuiModel>
  )
}

ModelDialog.defaultProps = {
  title: undefined,
  isModalOpen: false,
  closeModal: undefined,
}

export default withStyles(styles)(ModelDialog)
