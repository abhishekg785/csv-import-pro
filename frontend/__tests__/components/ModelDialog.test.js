import * as React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import ModelDialog from '../../src/components/ModelDialog'

configure({ adapter: new Adapter() })

describe('ModelDialog', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<ModelDialog />);
    expect(wrapper).toMatchSnapshot();
  })
})
