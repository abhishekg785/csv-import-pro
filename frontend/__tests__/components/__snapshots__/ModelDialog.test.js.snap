// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelDialog renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(ModelDialog) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "paper": "ModelDialog-paper-1",
      },
      "closeModal": undefined,
      "isModalOpen": false,
      "title": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "paper": "ModelDialog-paper-1",
        },
        "closeModal": undefined,
        "isModalOpen": false,
        "title": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
